dashboard/api.py

```diff
@@ -492,7 +493,16 @@ def is_exam_schedulable(user, course):
     """
     Check if a course is ready to schedule an exam or not
     """
-    schedulable_exam_runs = ExamRun.get_currently_schedulable(course)
+    now = now_in_utc()
+    # pick current and future runs
+    filter_query = Q(  # current exam runs
+        date_first_schedulable__lte=now,
+        date_last_schedulable__gte=now
+    )
+    filter_query.add(Q(date_last_eligible__gte=now.date()), Q.OR)  # future exam runs
```

Comment:
I don't understand what this query clause is doing is doing

Reply:
I think lines 498-501 means that we are filtering down to exams that are schedulable (removing ones that are outside the range of schedulable dates). 

But I don't understand what line 502 is doing. What effect does the OR have? And why do we need to include exams that are future eligible?

Reply:
line 502 is extracting future run just like https://github.com/mitodl/micromasters/blob/master/dashboard/utils.py#L429. So that if those future runs are authorize then count them in.

This the the reasons production user is suffering. He has a future authorized run which is not schedulable.